plugins/fused-ops-and-kernels/src/fms_acceleration_foak/models/granite.py

```diff
@@ -0,0 +1,130 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+from functools import partial
+
+# Third Party
+from fms_acceleration.model_patcher import (
+    ModelPatcherRule,
+    ModelPatcherTrigger,
+    combine_functions,
+    combine_triggers,
+)
+from transformers.models.granite.modeling_granite import (
```

Comment:
this needs to be gaurded because it only exists for later enough transformer versions
1. move this inside `get_mp_rules`
2. gaurd it inside a try catch
3. if `except`, then return and empty `[]`